enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,574 @@
+
+Furthermore, a lot of the rules about the structure of an index are implicitly documented in various sources that are scattered. A best effort is being made to keep the rules updated in the form of a library in the [api repo](https://github.com/operator-framework/api/tree/master/pkg/validation) that is imported by tools like `opm`, `operator-courier` etc that perform index validation. However, like all tools that aspire to provide data validation, these tools, or api libraries that these tools import the rules from, have to be constantly maintained to keep up with the latest rule changes, while also maintaining older rules in previous releases. This introduces a lot of overhead in ensuring that olm is easy to use by users, while also leaving windows for lapses due to mis-coordination between different tools across different versions.
+
+Since indexes are manipulated by making additive changes to it, the way an index can be reproduced today is my managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) along with a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, and as a by product has significantly burdened the `opm` tool with feature requests to add an increasing number of sub-commands to allow index authors to manipulate their indexes in various ways. 
+
+### Goals
+
+A human consumable representation of packages in an index in the form of json/yaml will: 
+
+1. Allow for editing of package level information without rebuilding and republishing artifacts(remove bundle/s from a channel, switch bundle/s to a different channel, edit the upgrade graph in a channel etc).
+2. Allow algebraic operations to be performed on indexes in a seamless way(combine two indexes, split an index, remove bundles from the index, etc).
+3. Allow for validation of the structure of an index against a standard definition of an index.
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducibility of an index using just the yaml/json representation of the index. 
+
+### Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations.    
+
+### User Stories 
```

Comment:
In POV the User stories should be really user stories. I mean, it ought to be : `As a <role> I can <capability>, so that <receive benefit>`

See; https://en.wikipedia.org/wiki/User_story

Then, what you added here shows to be more appropriate to `Implementation Details/Notes/Constraints [optional]`

Reply:
@camilamacedo86 are these stories [1](https://github.com/operator-framework/enhancements/pull/37/files/9948060747ac4cd336add31044b75ed55376e1f4#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R64),[2](https://github.com/operator-framework/enhancements/pull/37/files/9948060747ac4cd336add31044b75ed55376e1f4#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R169) ,[3](https://github.com/operator-framework/enhancements/pull/37/files/9948060747ac4cd336add31044b75ed55376e1f4#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R227) etc in a different format than what you mentioned?

Reply:
Yes. By following the standard we can clarify who and which circumstances. For example: 
> Template: As a <role> I can <capability>, so that <receive benefit>
Your user story here: 

> As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the packages of an index.
> The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  

Then, 

- What is the kind of the user which has `As a user with pull permission from the namespace an index is hosted`? What is its role in the context? 
- Then, see that if you add `so that ...` clarifies why it is important `I can query an index for a json/yaml representation of the packages of an index.`. What is the advantage?